lib/trace-writer.js

```diff
@@ -54,8 +57,59 @@ function TraceWriter(logger, config) {
     if (err) { return; } // ignore as index.js takes care of this.
     that.scheduleFlush_(project);
   });
+
+  that.getHostname(function(hostname) {
+    that.getInstanceId(function(instanceId) {
+      var labels = {};
+      labels[traceLabels.AGENT_DATA] = 'node ' + pjson.version;
+      labels[traceLabels.GCE_HOSTNAME] = hostname || require('os').hostname();
+      if (instanceId) {
+        labels[traceLabels.GCE_INSTANCE_ID] = instanceId;
+      }
+      var moduleName = process.env.GAE_MODULE_NAME || hostname;
+      labels[traceLabels.GAE_MODULE_NAME] = moduleName;
+
+      var moduleVersion = process.env.GAE_MODULE_VERSION;
+      if (moduleVersion) {
+        labels[traceLabels.GAE_MODULE_VERSION] = moduleVersion;
+        var minorVersion = process.env.GAE_MINOR_VERSION;
+        if (minorVersion) {
+          var versionLabel = '';
```

Comment:
It would be nice to move this craziness into a helper function for easier reasoning and so it can be easily eliminated once it is no longer necessary (which hopefully will be soon).